code/_onclick/hud/screen_objects.dm

```diff
@@ -106,54 +106,59 @@
+					H.internal = H.l_store
+				else if(istype(H.r_store, /obj/item/weapon/tank))
+					H << "<span class='notice'>You are now running on internals from the [H.r_store] in your right pocket.</span>"
+					H.internal = H.r_store
+
+			//Seperate so CO2 jetpacks are a little less cumbersome.
+			if(!C.internal && istype(C.back, /obj/item/weapon/tank))
+				C << "<span class='notice'>You are now running on internals from the [C.back] on your back.</span>"
+				C.internal = C.back
+
 			if(C.internal)
-				C.internal = null
-				C << "<span class='notice'>You are no longer running on internals.</span>"
-				icon_state = "internal0"
+				icon_state = "internal1"
 			else
-				if(!istype(C.wear_mask, /obj/item/clothing/mask))
-					C << "<span class='warning'>You are not wearing an internals mask!</span>"
-					return 1
-				else
-					var/obj/item/clothing/mask/M = C.wear_mask
-					if(M.mask_adjusted) // if mask on face but pushed down
-						M.adjustmask(C) // adjust it back
-					if( !(M.flags & MASKINTERNALS) )
-						C << "<span class='warning'>You are not wearing an internals mask!</span>"
-						return
-					if(istype(C.l_hand, /obj/item/weapon/tank))
-						C << "<span class='notice'>You are now running on internals from the [C.l_hand] on your left hand.</span>"
-						C.internal = C.l_hand
-					else if(istype(C.r_hand, /obj/item/weapon/tank))
-						C << "<span class='notice'>You are now running on internals from the [C.r_hand] on your right hand.</span>"
-						C.internal = C.r_hand
-					else if(ishuman(C))
-						var/mob/living/carbon/human/H = C
-						if(istype(H.s_store, /obj/item/weapon/tank))
-							H << "<span class='notice'>You are now running on internals from the [H.s_store] on your [H.wear_suit].</span>"
-							H.internal = H.s_store
-						else if(istype(H.belt, /obj/item/weapon/tank))
-							H << "<span class='notice'>You are now running on internals from the [H.belt] on your belt.</span>"
-							H.internal = H.belt
-						else if(istype(H.l_store, /obj/item/weapon/tank))
-							H << "<span class='notice'>You are now running on internals from the [H.l_store] in your left pocket.</span>"
-							H.internal = H.l_store
-						else if(istype(H.r_store, /obj/item/weapon/tank))
-							H << "<span class='notice'>You are now running on internals from the [H.r_store] in your right pocket.</span>"
-							H.internal = H.r_store
-
-					//Seperate so CO2 jetpacks are a little less cumbersome.
-					if(!C.internal && istype(C.back, /obj/item/weapon/tank))
-						C << "<span class='notice'>You are now running on internals from the [C.back] on your back.</span>"
-						C.internal = C.back
-
-					if(C.internal)
-						icon_state = "internal1"
-					else
-						C << "<span class='warning'>You don't have an oxygen tank!</span>"
+				C << "<span class='warning'>You don't have an oxygen tank!</span>"
+				return
+	C.update_action_buttons_icon()

```

Comment:
this is the only real change in this file, we update the action buttons icon (since the internal tank button turns green when used as internals now).